contracts/evoting/types/ballots.go

```diff
@@ -512,6 +494,46 @@ func (r Rank) GetChoicesLength() int {
 	return len(r.Choices)
 }
 
+// unmarshalAnswers interprets the given raw answers into a slice of integer
+// representing the ranking of each choice and ensures the answers are correctly
+// formatted
+func (r Rank) unmarshalAnswers(ranks []string) ([]int8, error) {
+	if len(ranks) != len(r.Choices) {
+		return nil, fmt.Errorf("question %s has a wrong number of answers:"+
+			" expected %d got %d", r.ID, len(r.Choices), len(ranks))
+	}
+
+	var selected uint = 0
+	results := make([]int8, 0)
```

Comment:
This prevents re-allocating a new slice for each `append`:

```suggestion
	results := make([]int8, 0, len(ranks))
```

then you can directly set `results[i] = int8(rankValue)`